packages/in_app_purchase/lib/src/in_app_purchase/in_app_purchase_connection.dart

```diff
@@ -81,14 +81,18 @@ abstract class InAppPurchaseConnection {
   /// You always need to restore all the non consumable products for user when
   /// they switch their phones.
   ///
-  /// This method does not return anything. Instead, after triggering this
-  /// method, purchase updates will be sent to [purchaseUpdatedStream]. You
-  /// should [Stream.listen] to [purchaseUpdatedStream] to get [PurchaseDetails]
-  /// objects in different [PurchaseDetails.status] and update your UI
-  /// accordingly. When the [PurchaseDetails.status] is
-  /// [PurchaseStatus.purchased] or [PurchaseStatus.error], you should deliver
-  /// the content or handle the error. On iOS, you also need to call
-  /// [completePurchase] to finish the purchasing process.
+  /// This method does not return the result of the purchase. Instead, after
+  /// triggering this method, purchase updates will be sent to
+  /// [purchaseUpdatedStream]. You should [Stream.listen] to
+  /// [purchaseUpdatedStream] to get [PurchaseDetails] objects in different
+  /// [PurchaseDetails.status] and update your UI accordingly. When the
+  /// [PurchaseDetails.status] is [PurchaseStatus.purchased] or
+  /// [PurchaseStatus.error], you should deliver the content or handle the
+  /// error. On iOS, you also need to call [completePurchase] to finish the
+  /// purchasing process.
+  ///
+  /// This method does return whether or not the purchase request was initially
```

Comment:
The purchases going to the Stream instead of being returned in an immediate Future is already fairly unidiomatic for Dart, even though I do think it makes the most sense here given our options. However I'm worried that this change is going to make it even more confusing, since now this method _would_ be returning a `Future`... but not a `Future` that actually has to do with the final purchase outcome, just whether the initial request to show a purchase dialog to the user went through. I'm not sure what the right call to try and expose that extra info without being extra confusing is here.

Reply:
I think it is necessary to have it return a BOOL. It looks good with the updated documentation.
The other alternative that I can think of is to create a unified response code to specify the response just like Android but I feel like it would make things more confusing.